app/models/nic/base.rb

```diff
@@ -210,16 +230,31 @@ def sync_name
       synchronizer.sync_name if synchronizer.sync_required?
     end
 
-    def validate_host_location
+    def validate_host_subnet_location
       return true if self.host.location.nil? && self.subnet.locations.empty?
       errors.add(:subnet, _("is not defined for host's location.")) unless include_or_empty?(self.subnet.locations, self.host.location)
     end
 
-    def validate_host_organization
+    def validate_host_subnet_organization
       return true if self.host.organization.nil? && self.subnet.organizations.empty?
       errors.add(:subnet, _("is not defined for host's organization.")) unless include_or_empty?(self.subnet.organizations, self.host.organization)
     end
 
+    def validate_host_subnet6_location
```

Comment:
Same as above, take the whole `validate_host_subnetx_xxxxx` and define in terms of `Taxonomy.enabled_taxonomies` please